src/ParametersHandler/tests/ParametersHandlerYarpTest.cpp

```diff
@@ -124,12 +124,13 @@ TEST_CASE("Get parameters")
 
     SECTION("is Empty")
     {
-        IParametersHandler::shared_ptr groupHandler = parameterHandler->getGroup("CARTOONS").lock();
-        REQUIRE_FALSE(groupHandler);
+        REQUIRE_FALSE(parameterHandler->getGroup("CARTOONS").lock());
         IParametersHandler::shared_ptr setGroup = std::make_shared<YarpImplementation>();
         REQUIRE(parameterHandler->setGroup("CARTOONS", setGroup));
 
-        groupHandler = parameterHandler->getGroup("CARTOONS").lock(); //now the pointer should be lockable
+        IParametersHandler::shared_ptr groupHandler
+            = parameterHandler->getGroup("CARTOONS").lock()->clone(); // now the pointer should be
```

Comment:
Same as above, although it may also make sense to test the ``const`` version of the method. So some of these ``clone`` may remain.